sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+
+namespace Azure.Communication
+{
+    internal class CommunicationIdentifierSerializer
+    {
+        public static CommunicationIdentifier Deserialize(CommunicationIdentifierModel identifier)
+        {
+            var id = identifier.Id;
+            var kind = identifier.Kind;
+
+            if (kind == CommunicationIdentifierKind.CommunicationUser)
```

Comment:
Use 'switch' instead of if here?

Reply:
Hopefully in future C# we can do that. Considering that this not a real enum, hence not compile time constant, we cannot use `switch` 😥